rust/src/compose.rs

```diff
@@ -687,10 +701,23 @@ impl RootfsOpts {
                     .iter()
                     .flat_map(|v| ["--source-root", v.as_str()]),
             )
-            .args([manifest.as_str(), self.dest.as_str()])
+            .args([manifest.as_str(), dest_path.as_str()])
             .run()
             .context("Executing compose install")?;
 
+        // If we used a cachedir then we stored the target destination rootfs
+        // under the cachedir. Let's copy it out now. In this case we copy it
+        // out but we explicitly break reflinks because if we don't the
+        // fixup_installroot below will render the cache under cachedir useless
+        // (i.e. the changes to the files that would still be reflinked to files
+        // in cachedir would cause the cache to be unusable on the next run).
+        if self.cachedir.is_some() {
+            Command::new("cp")
+                .args(["-a", "--reflink=never"])
+                .args([dest_path.as_str(), self.dest.as_str()])
+                .run()?;
```

Comment:
let me know if there is a library function that we should prefer to use here for something like a recursive copy.